### PR TITLE
PR Taisync support for Android

### PR DESCRIPTION
**Pull request by MatejFranceskin**
Ported TaiSync Support to Android platform.

### CODE REVIEW

File: android/res/xml/device_filter.xml [hunk 1]
```diff
@@ -2,5 +2,6 @@
 <resources>
     <!-- Allow anything connected -->
     <usb-device />
+    <usb-accessory />
```
Comment (dogmaphobic) [inline android/res/xml/device_filter.xml]: Is there a way to limit this to the Taisync? The Android documentation says you should though I can't quite figure out what information you would provide that would _filter_ the Taisync only:

_In this case, the following resource file should be saved in res/xml/accessory_filter.xml and specifies that any accessory that has the corresponding model, manufacturer, and version should be filtered. The accessory sends these attributes the Android-powered device:_
```
<?xml version="1.0" encoding="utf-8"?>

<resources>
    <usb-accessory model="DemoKit" manufacturer="Google" version="1.0"/>
</resources>
```

Comment (MatejFranceskin) [reply android/res/xml/device_filter.xml]: Yes indeed - I limited it to:
<usb-accessory model="android.usbaoa" manufacturer="taisync" version="1.0"/>

**Comment by DonLakeFlyer** [comment]
My concern with this is that it doesn't conflict with usb radios/boards being plugged in. I just went througha nightmare change to hopefully fix #5383. My concerns is that this changes regular usb device behavior as well. Is there any overlap there with respect to ```usb-device``` showing up as ```usb-accesory``` as well?

**Comment by MatejFranceskin** [comment]
Found a bug in the implementation - it recognizes USB accessory only when device is in developer or debug mode. Please wait for the fix and more testing.

**Comment by MatejFranceskin** [comment]
> Found a bug in the implementation - it recognizes USB accessory only when device is in developer or debug mode. Please wait for the fix and more testing.

It turns out that this is how it is implemented and cannot be changed:
Currently android usb multimedia transmission requires the support of aoa protocol, there are currently two versions 0x2D00 and 0x2D01. If android terminal aoa version is 0x2D01 must enable usb debuging mode. http://www.android-doc.com/tools/adk/aoa2.html.
Android devices that we tested use 0x2D01 version and require USB debugging to be switched on.

**Comment by MatejFranceskin** [comment]
> My concern with this is that it doesn't conflict with usb radios/boards being plugged in. I just went througha nightmare change to hopefully fix #5383. My concerns is that this changes regular usb device behavior as well. Is there any overlap there with respect to `usb-device` showing up as `usb-accesory` as well?

I don't think that this implementation interferes with the was USB devices are handled. I tested it with direct USB connection to AP and it works like it always did.

**Pull request merged**